libparsec/crates/client_connection/src/authenticated_cmds/mod.rs

```diff
@@ -85,16 +90,34 @@ impl AuthenticatedCmds {
 
         let request_body = request.api_dump()?;
 
+        let url = match T::FAMILY {
+            ProtocolFamily::Authenticated => self.url.clone(),
+            ProtocolFamily::Tos => {
+                let mut url = self.url.clone();
+                url.path_segments_mut()
+                    .expect("Url is not cannot-be-a-base")
+                    .push("tos");
+                url
+            }
+            // Other families are not supported
+            _ => {
+                // TODO: Replace this runtime error by a const panic once available
+                // (see https://rust-lang.github.io/rfcs/2345-const-panic.html)
```

Comment:
We are not in a `const function` so a `const_panic` would not help here.

```suggestion
```

Reply:
Can you include that part in the comment then ? for a future rewrite

Reply:
I find it self explanatory as it is, but if you insist ;-)

done.